views/js/runner/plugins/content/accessibility/keyNavigation.js

```diff
@@ -379,43 +440,72 @@ define([
      * @param testRunner
      * @returns {*}
      */
-    function initTestRunnerNavigation(testRunner){
-
+    function initTestRunnerNavigation(testRunner, state){
+        var keyNavigatorItem;
         var navigators;
 
         //blur current focused element, to reinitialize keyboard navigation
         if (document.activeElement){
             document.activeElement.blur();
         }
 
-        navigators = _.union(
-            initRubricNavigation(testRunner),
-            initContentNavigation(testRunner),
-            initToolbarNavigation(testRunner),
-            initNavigatorNavigation(testRunner),
-            initHeaderNavigation(testRunner)
-        );
+        switch (state.contentNavigatorType) {
+            case 'linear' :
+                navigators = _.union(
+                    initRubricNavigation(testRunner),
+                    initAllContentButtonsNavigation(testRunner),
+                    initToolbarNavigation(testRunner),
+                    initNavigatorNavigation(testRunner),
+                    initHeaderNavigation(testRunner),
+                );
+            break;
+
+            default:
+                navigators = _.union(
+                    initRubricNavigation(testRunner),
+                    initDefaultContentNavigation(testRunner),
+                    initToolbarNavigation(testRunner),
+                    initNavigatorNavigation(testRunner),
+                    initHeaderNavigation(testRunner),
+                );
+            break;
+        }
+
 
         navigators = navigableGroupElement.createFromNavigators(navigators);
 
-        return keyNavigator({
+        keyNavigatorItem = keyNavigator({
             id : 'test-runner',
             replace : true,
             loop : true,
             elements : navigators
-        }).on('tab', function(elem){
-            if (!allowedToNavigateFrom(elem)) {
-                return false;
-            } else {
-                this.next();
+        });
+
+        keyNavigatorItem.on('tab', function(elem){
+            if (allowedToNavigateFrom(elem)) {
+              this.next();
             }
         }).on('shift+tab', function(elem){
-            if (!allowedToNavigateFrom(elem)) {
-                return false;
-            } else {
-                this.previous();
+            if (allowedToNavigateFrom(elem)) {
+              this.previous();
             }
         });
+
+        if ( state.contentNavigatorType === 'linear' ) {
+            keyNavigatorItem.on('right', function(elem){
+
+                var isCurrentElementFirst = $(elem).is(':first-child');
+
+                isCurrentElementFirst && allowedToNavigateFrom(elem) && this.next();
```

Comment:
Prefer to use formal `if..else` structure. Shorten logicals are only allowed in value initializations.

Reply:
done